bread/layout/base.py

```diff
@@ -1,10 +1,32 @@
 import htmlgenerator as hg
+from django.conf import settings
+from django.http import HttpResponse
+from django.template.context import _builtin_context_processors
+from django.utils.module_loading import import_string
 
 from bread.utils import pretty_modelname, resolve_modellookup
 from bread.utils.urls import reverse_model
 
 from ..formatters import format_value
 
+CONTEXT_PROCESSORS = tuple(
+    import_string(path)
+    for path in _builtin_context_processors
+    + tuple(settings.TEMPLATES[0]["OPTIONS"]["context_processors"])
+)
+
+
+class HasBreadCookieValue(hg.Lazy):
+    def __init__(self, cookiename, value):
+        self.cookiename = cookiename
+        self.value = value
+
+    def resolve(self, context, element):
+        return (
+            context["request"].session["bread-cookies"].get(f"bread-{self.cookiename}")
+            == self.value
+        )
+
```

Comment:
We can use this lazy element (e.g. together with ```hg.If```) in order to render output depending on a certain cookie state. Only ```"bread-"``` prefixed cookies are considered. This is thought to be the standard way to manage client-based state which affects rendering of the page on the server side.